tip-distributor/src/claim_mev_workflow.rs

```diff
@@ -70,71 +82,60 @@ pub fn claim_mev_tips(
-                                let ix = Instruction {
-                                    program_id: *tip_distribution_program_id,
-                                    data: tip_distribution::instruction::Claim {
-                                        proof: node.proof.unwrap(),
-                                        amount: node.amount,
-                                        bump: claim_status_bump,
-                                    }
-                                    .data(),
-                                    accounts: tip_distribution::accounts::Claim {
-                                        config: tip_distribution_config,
-                                        tip_distribution_account: tree.tip_distribution_account,
-                                        claimant: node.claimant,
-                                        claim_status: claim_status_pubkey,
-                                        payer: keypair.pubkey(),
-                                        system_program: system_program::id(),
-                                    }
-                                    .to_account_metas(None),
-                                };
-                                let transaction = Transaction::new_signed_with_payer(
-                                    &[ix],
-                                    Some(&keypair.pubkey()),
-                                    &[&keypair],
-                                    blockhash,
-                                );
-                                info!("tx: {:?}", transaction);
-                                transactions.push(transaction);
-                            } else {
-                                panic!("unexpected rpc error: {:?}", e);
-                            }
-                        }
-                    }
-                } else {
+                let fetched_tip_distribution_account = TipDistributionAccount::try_deserialize(&mut account.data.as_slice()).expect("failed to deserialize tip_distribution_account state");
+                if fetched_tip_distribution_account.merkle_root.is_none() {
                     info!(
-                        "not claiming because merkle root isn't uploaded yet claimant: {:?} tda: {:?}",
+                        "not claiming because merkle root isn't uploaded yet. claimant: {:?} tda: {:?}",
                         node.claimant,
                         tree.tip_distribution_account
                     );
+                    continue;
+                }
+
+                let claim_status_acc_result = rpc_client.get_account(&claim_status_pubkey).await;
+                if claim_status_acc_result.is_ok() {
+                    debug!("claim status account already exists: {:?}", claim_status_pubkey);
+                    continue;
                 }
+                if !(matches!(claim_status_acc_result.unwrap_err().kind(), ErrorKind::RpcError(RpcError::ForUser(_)))) {
+                    panic!("Invalid RPC Error");
```

Comment:
wdyt about matching on the error and printing it here?

Reply:
can log claim_status_acc_result here

Reply:
logged now